integration/common.go

```diff
@@ -17,33 +17,39 @@ import (
 	"github.com/stretchr/testify/suite"
 )
 
-type fileUploadSuite struct {
+// FileUploadSuite tests the file upload functionalities, using a provided configuration
+type FileUploadSuite struct {
 	suite.Suite
 	util.SuiteInitializer
 
-	thingURL   string
-	featureURL string
-	uploadCfg  uploadTestConfig
+	ThingURL   string
+	FeatureURL string
+
+	uploadCfg UploadTestConfig
 }
 
 type httpFileUploadSuite struct {
-	fileUploadSuite
+	FileUploadSuite
 }
 
 type azureFileUploadSuite struct {
-	fileUploadSuite
+	FileUploadSuite
 }
 
 type awsFileUploadSuite struct {
-	fileUploadSuite
+	FileUploadSuite
 }
 
-type uploadTestConfig struct {
+// UploadTestConfig contains the test configuration data, such as upload directory and http server URL
+type UploadTestConfig struct {
```

Comment:
Double check the struct visibility. It seems that it's not required to be public.